scripts/pylib/twister/twisterlib.py

```diff
@@ -2809,15 +2840,19 @@ def get_all_tests(self):
 
     @staticmethod
     def get_toolchain():
-        toolchain = os.environ.get("ZEPHYR_TOOLCHAIN_VARIANT", None)
+        toolchain_script = ZEPHYR_BASE + "/cmake/verify-toolchain.cmake"
+        result = CMake.run_cmake_script([toolchain_script, "FORMAT=json"])
+
         try:
-            if not toolchain:
+            if result['returncode']:
                 raise TwisterRuntimeError("E: Variable ZEPHYR_TOOLCHAIN_VARIANT is not defined")
         except Exception as e:
             print(str(e))
             sys.exit(2)
 
-        return toolchain
+        toolchain_info = json.loads(result['stdout'])
+
```

Comment:
enhancement, just print out what toolchain is being used with logger.info(..) here.

Reply:
not sure I understand.
`logger.info(...)` would print something, but we are not printing anything here.

The `toolchain_info` is a dictionary containing the output from `verify-toolchain.cmake` looking like:
```
{"ZEPHYR_TOOLCHAIN_VARIANT" : "zephyr",
 "SDK_VERSION": "0.12.0",
 "ZEPHYR_SDK_INSTALL_DIR" : "/opt/zephyr-sdk-0.12.0-beta"}
```
and we are creating a dict based on this data.

This mean we return the value of `ZEPHYR_TOOLCHAIN_VARIANT` to rest of `twister` which is needed for example here:
https://github.com/zephyrproject-rtos/zephyr/blob/ec3748d6b0988a7f8bd9e845e54ea54aecba4aae/scripts/pylib/twister/twisterlib.py#L2975
so that test cases can be correctly filtered based on the toolchain.

Reply:
I am saying, display the information about which toolchain is being used, so

logger.info(f"Using {toolchain_info.get('ZEPHYR_TOOLCHAIN_VARIANT')} toolchain.")

this is just informational, given that the toolchain now will be detected automatically if no variant is set.

Reply:
Good point. Done.